### PR TITLE
spaces are allowed in userids

### PR DESCRIPTION
**Pull request by szaimen**
Address https://github.com/nextcloud/nextcloud-vue/pull/3468#issuecomment-1325068316

Signed-off-by: Simon L <szaimen@e.mail.de>

### CODE REVIEW

File: lib/private/User/Manager.php [hunk 1]
```diff
@@ -428,10 +428,10 @@ public function createUserFromBackend($uid, $password, UserInterface $backend) {
 		$l = \OC::$server->getL10N('lib');
 
 		// Check the name for bad characters
-		// Allowed are: "a-z", "A-Z", "0-9" and "_.@-'"
+		// Allowed are: "a-z", "A-Z", "0-9", spaces and "_.@-'"
 		if (preg_match('/[^a-zA-Z0-9 _.@\-\']/', $uid)) {
 			throw new \InvalidArgumentException($l->t('Only the following characters are allowed in a username:'
-				. ' "a-z", "A-Z", "0-9", and "_.@-\'"'));
+				. ' "a-z", "A-Z", "0-9", spaces and "_.@-\'"'));
```
Comment (nickvergessen) [inline lib/private/User/Manager.php:434]:
```suggestion
				. ' "a-z", "A-Z", "0-9", spaces, and "_.@-\'"'));
```

**Review by nickvergessen — APPROVED** [review]

File: apps/user_ldap/lib/Access.php [hunk 2]
```diff
@@ -1362,11 +1362,8 @@ public function sanitizeUsername($name) {
 		// Remove unknown leftover entities
 		$name = preg_replace('#&[^;]+;#', '', $name);
 
-		// Replacements
-		$name = str_replace(' ', '_', $name);
-
 		// Every remaining disallowed characters will be removed
-		$name = preg_replace('/[^a-zA-Z0-9_.@-]/u', '', $name);
+		$name = preg_replace('/[^a-zA-Z0-9 _.@-]/u', '', $name);
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Access.php:1366]: not sure about that one. If ldap never allowed it there could be other reasons.
Also this now breaks existing users?

Comment (come-nc) [reply apps/user_ldap/lib/Access.php:1366]: Only unmapped users will trigger a call to sanitizeUsername so no problem for existing users.
But I still find it unexpected that user ids with spaces are generated. But if the rest of the software allows it, I suppose there is no reason for LDAP to forbid it? Default configuration is UUID anyway.

**Comment by szaimen** [comment]
I ran into an issue when trying to rebase, cherry-picked and create a new PR: https://github.com/nextcloud/server/pull/36067

**Pull request closed without merging**